keps/sig-cli/0031-kustomize-integration.md

```diff
@@ -0,0 +1,230 @@
+```
+kubectl apply -f <dir> --dry-run -o yaml|json
+```
+
+#### get
+The command visible to users is
+```
+kubectl get -f <dir>
+```
+To get the detailed objects in a kustomization
+```
+kubectl get -f <dir> --dry-run -o yaml|json
+```
+
+#### delete
+The command visible to users is
+```
+kubectl delete -f <dir>
+```
+
+### Implementation Details/Notes/Constraints
+
+To enable kustomize in kubectl, the function `FilenameParam` inside Builder type will be updated to recognize kustomization directories. The Builder will expand the sources in a kustomization directory and pass them to a subcommand.
+
+ Since kustomization directories themselves have a recursive structure, `-R` will be ignored on those directories. Allowing recursive visit to the same files will lead to duplicate resources. 
+
+The examples and descriptions for apply, get and delete will be updated to include the support of kustomization directories. 
+
+### Risks and Mitigations
+
+This KEP doesn't provide a editing command for kustomization.yaml file. Users will either manually edit this file or use `Kusotmize edit`. 
+
+## Graduation Criteria
+
+There are two signals that can indicate the success of this integration.
+- Kustomize users drop the piped commands `kustomize build <dir> | kubectl apply -f - ` and start to use apply directly.
+- Kubectl users put their configuration files in kustomization directories.
+
+
+## Implementation History
+
+Most implementation will be in cli-runtime
+
+- vendor `kustomize/pkg` into kubernetes
+- copy `kustomize/k8sdeps` into cli-runtime
+- Implement a Visitor for kustomization directory which
+   - execute kustomize build to get a list of resources
+   - write the output to a StreamVisitor
+- When parsing filename parameters in FilenameParam, look for kustomization directories
+- update the examples in kubectl commands
+- Improve help messages or documentations to list kubectl subcommands that can work with kustomization directories
+
+## Alternatives
+
+The approaches in this section are considered, but rejected.
+### Copy kustomize into staging
+Copy kustomize code into kubernetes/staging and have the staging kustomize as source of truth. The public kustomize repo will be synced automatically with the staging kustomize.
+- Pros
+  - Issues can be managed in the public repo
+  - The public repo can provide a kustomize binary
+  - The public repo can be used as kustomize libraries   
+  - Empower further integration of kubectl with kustomize
+- Cons
+  - The staging repo is designed for libraries that will be separated out. Moving kustomize into staging sounds controversial
+  - Kustomize will be in staging, the development will be done in k/k repository
+  - Development velocity will be reduced of every release
+
+### Add kustomize as a subcommand in kubectl
+Add kustomize as a subcommand into kubectl was the first way we tried to enable kustomize in kubectl. The PR was PR: add kustomize as a subcommand of kubectl
```

Comment:
`The PR was PR:` -> possibly reword or quote.

Reply:
Added quote to the PR.